### DISCUSSION TITLE
出现日志读取模块出错

### DISCUSSION THREAD

**Discussion by guwukawa** [sunfkny/genshin-gacha-export#40] [提问 / Q&A] [answered]
系统版本：
windows10 专业版 21H2 19044.1466
Windows Feature Experience Pack
错误内容如下
日志读取模块出错:
 Traceback (most recent call last):
  File "urllib3\connection.py", line 169, in _new_conn
  File "urllib3\util\connection.py", line 96, in create_connection
  File "urllib3\util\connection.py", line 86, in create_connection
TimeoutError: [WinError 10060] 由于连接方在一段时间后没有正确答复或连接的主机没有反应，连接尝试失败。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "urllib3\connectionpool.py", line 699, in urlopen
  File "urllib3\connectionpool.py", line 382, in _make_request
  File "urllib3\connectionpool.py", line 1010, in _validate_conn
  File "urllib3\connection.py", line 353, in connect
  File "urllib3\connection.py", line 181, in _new_conn
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x000001E6CEDA64C0>: Failed to establish a new connection: [WinError 10060] 由于连接方在一段时间后没有正确答复或连接的主机没有反应，连接尝试失败。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "requests\adapters.py", line 439, in send
  File "urllib3\connectionpool.py", line 755, in urlopen
  File "urllib3\util\retry.py", line 573, in increment
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='hk4e-api.mihoyo.com', port=443): Max retries exceeded with url: /event/gacha_info/api/getGachaLog?authkey_ver=1&sign_type=2&auth_appid=webview_gacha&init_type=301&gacha_id=9a9e9cf2b6978c85fe9f6bea31f7826d4d54a2&timestamp=1641338897&lang=zh-cn&device_type=pc&ext=%7B%22loc%22%3A%7B%22x%22%3A-476.3782958984375%2C%22y%22%3A210.6436767578125%2C%22z%22%3A330.84381103515627%7D%2C%22platform%22%3A%22WinST%22%7D&game_version=CNRELWin2.4.0_R5578371_S5635986_D5598173&plat_type=pc&region=cn_gf01&authkey=O7lOOv8JUMCje8be10Qtfk%2FbOcb7KvKVzi2dFMgkYjkq4tuk%2BMneQjRIgzvWQm6bHQzbhIWuoLgNamvYhD2FvKhP9bDLyTf9RwkeFDxjT2oJpOsEEEMpeH5baIA4HfuJy%2FGMnPk0b7gXdFV9WRar2XxI%2BPal9tn%2BiZ%2B9vhDGHF%2FE4%2F4CRzgBSxBUnYpRRZYHSKst3CWOSGKiJuGJDWBsbE3wCap%2FS6TeH%2F%2Fj7sgc3Vib1dZW0qUcgA7wPrpRIAyQuj4oqgHGxqJ7XdTxOs%2BwLXuQKBVSqEUGShTvVliWh5etRk6NGDa8Kwo1tJGFbCOdWK%2FvVkadSXN%2FWjosVpq1xaVDQbrTBEeuYSFZJsXEeuXPK%2BChoFI3OZoY9ZHa7cyrpMI3MVyDEUw5mUqtXl%2Fu30depAsglOSs0aLYlqD7EV4mPn4u11mIhi260VA38mSi8YCauCYYASqWL3wF2CdguClzRsfqVOu0CPEJEfgyYXWqe0680f7J7Bpxa1A0kIbQrQVwctIq%2FHrf3JpjNkZz3WxgTxhczGWue4DA2VASFLXmWamnzHNxhTxkGagVHGBTuSxeod7eNrS%2FLR%2F8iOWWhjX3Kl1fofLR5rQU9n4K%2F3g3mEGNMutp1U5jwnTP%2B3%2B19uyyho7fwVsNghP%2Fd%2BK5eAJwtcjb9VijjGdhfUBsVrvPJVukPxfCg7KRpDcxndFS0iFHQ%2FnahAS9V%2F8ztYAFRB4u68c%2F8JaX12H8VtW5LQgnskRNOOMiKwETNGacRY58c1QQG8D56YIgTMWDZF%2FmuWQstrpPxec%2Fj%2BM71sDOmfyGeTGfRMYp7R3MXcSr%2BRu7rA0mj1Z28V%2B1AXjPfMSzoijy6ePkApzdVFDbQ29i%2B3tEPAmz76cOZqUkk%2Fg0TBDKGqEBPPrq7aTDqqu0T3ff0AGQTQExPVvh7k3mD%2FMxTxP5dy9zimM6EJbJ%2BMPreQ0MOBSNwJWjCtHPC7tCCE6sgU0Mg7VgB0qlYMMgvDnQW2%2FayIpAAg%2F5bVbykis74RKyeBpLL85UN3TRbLnOsj%2FqmEakKNwjTwkr1OTkGzchfuVqEvL5UCcAJSdqUja6pQC6KakfvR%2BHMlVZIJPLPHHpewwHz9K%2BMbhbKtMtGmQ6KVY8JpqhZRbZKIIVHLeUWEWjhYxbVjeM6qtX3hGk6FHJmZIxu5HxUFOLYC4Sy45F%2BPMrGjKLpueecdk%2FcRO3v6HQ61q7AOoWiPofHbaJcFn2L563hnKBlY5si5DNzTnGeiOjkxzQWd1l%2FJdjuJx8VjyIjxDQYVEjNfo8ZNmuglq3VDVEA%2FzNw3oFr23ZqvZVxje7lwzE0VQrw5JT2WaB66wT8EgccQrO0%2F3h0Gy%2FWT1MXA%3D%3D&game_biz=hk4e_cn&size=20&gacha_type=301&page=14&end_id=1642741560000896558 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x000001E6CEDA64C0>: Failed to establish a new connection: [WinError 10060] 由于连接方在一段时间后没有正确答复或连接的主机没有反应，连接尝试失败。'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "main.py", line 414, in <module>
    print("合法")
  File "main.py", line 30, in main
    gachaLog = getGachaLogs(gachaTypeId, gachaQueryTypeDict)
  File "main.py", line 160, in getGachaLogs
    r = requests.get(api)
  File "requests\api.py", line 76, in get
  File "requests\api.py", line 61, in request
  File "requests\sessions.py", line 542, in request
  File "requests\sessions.py", line 655, in send
  File "requests\adapters.py", line 516, in send
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='hk4e-api.mihoyo.com', port=443): Max retries exceeded with url: /event/gacha_info/api/getGachaLog?authkey_ver=1&sign_type=2&auth_appid=webview_gacha&init_type=301&gacha_id=9a9e9cf2b6978c85fe9f6bea31f7826d4d54a2&timestamp=1641338897&lang=zh-cn&device_type=pc&ext=%7B%22loc%22%3A%7B%22x%22%3A-476.3782958984375%2C%22y%22%3A210.6436767578125%2C%22z%22%3A330.84381103515627%7D%2C%22platform%22%3A%22WinST%22%7D&game_version=CNRELWin2.4.0_R5578371_S5635986_D5598173&plat_type=pc&region=cn_gf01&authkey=O7lOOv8JUMCje8be10Qtfk%2FbOcb7KvKVzi2dFMgkYjkq4tuk%2BMneQjRIgzvWQm6bHQzbhIWuoLgNamvYhD2FvKhP9bDLyTf9RwkeFDxjT2oJpOsEEEMpeH5baIA4HfuJy%2FGMnPk0b7gXdFV9WRar2XxI%2BPal9tn%2BiZ%2B9vhDGHF%2FE4%2F4CRzgBSxBUnYpRRZYHSKst3CWOSGKiJuGJDWBsbE3wCap%2FS6TeH%2F%2Fj7sgc3Vib1dZW0qUcgA7wPrpRIAyQuj4oqgHGxqJ7XdTxOs%2BwLXuQKBVSqEUGShTvVliWh5etRk6NGDa8Kwo1tJGFbCOdWK%2FvVkadSXN%2FWjosVpq1xaVDQbrTBEeuYSFZJsXEeuXPK%2BChoFI3OZoY9ZHa7cyrpMI3MVyDEUw5mUqtXl%2Fu30depAsglOSs0aLYlqD7EV4mPn4u11mIhi260VA38mSi8YCauCYYASqWL3wF2CdguClzRsfqVOu0CPEJEfgyYXWqe0680f7J7Bpxa1A0kIbQrQVwctIq%2FHrf3JpjNkZz3WxgTxhczGWue4DA2VASFLXmWamnzHNxhTxkGagVHGBTuSxeod7eNrS%2FLR%2F8iOWWhjX3Kl1fofLR5rQU9n4K%2F3g3mEGNMutp1U5jwnTP%2B3%2B19uyyho7fwVsNghP%2Fd%2BK5eAJwtcjb9VijjGdhfUBsVrvPJVukPxfCg7KRpDcxndFS0iFHQ%2FnahAS9V%2F8ztYAFRB4u68c%2F8JaX12H8VtW5LQgnskRNOOMiKwETNGacRY58c1QQG8D56YIgTMWDZF%2FmuWQstrpPxec%2Fj%2BM71sDOmfyGeTGfRMYp7R3MXcSr%2BRu7rA0mj1Z28V%2B1AXjPfMSzoijy6ePkApzdVFDbQ29i%2B3tEPAmz76cOZqUkk%2Fg0TBDKGqEBPPrq7aTDqqu0T3ff0AGQTQExPVvh7k3mD%2FMxTxP5dy9zimM6EJbJ%2BMPreQ0MOBSNwJWjCtHPC7tCCE6sgU0Mg7VgB0qlYMMgvDnQW2%2FayIpAAg%2F5bVbykis74RKyeBpLL85UN3TRbLnOsj%2FqmEakKNwjTwkr1OTkGzchfuVqEvL5UCcAJSdqUja6pQC6KakfvR%2BHMlVZIJPLPHHpewwHz9K%2BMbhbKtMtGmQ6KVY8JpqhZRbZKIIVHLeUWEWjhYxbVjeM6qtX3hGk6FHJmZIxu5HxUFOLYC4Sy45F%2BPMrGjKLpueecdk%2FcRO3v6HQ61q7AOoWiPofHbaJcFn2L563hnKBlY5si5DNzTnGeiOjkxzQWd1l%2FJdjuJx8VjyIjxDQYVEjNfo8ZNmuglq3VDVEA%2FzNw3oFr23ZqvZVxje7lwzE0VQrw5JT2WaB66wT8EgccQrO0%2F3h0Gy%2FWT1MXA%3D%3D&game_biz=hk4e_cn&size=20&gacha_type=301&page=14&end_id=1642741560000896558 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x000001E6CEDA64C0>: Failed to establish a new connection: [WinError 10060] 由于连接方在一段时间后没 有正确答复或连接的主机没有反应，连接尝试失败。'))

**Comment by sunfkny** [comment]
网络超时了，你可以尝试浏览器打开 [这个链接](https://hk4e-api.mihoyo.com/event/gacha_info/api/getGachaLog?authkey_ver=1&sign_type=2&auth_appid=webview_gacha&init_type=301&gacha_id=9a9e9cf2b6978c85fe9f6bea31f7826d4d54a2&timestamp=1641338897&lang=zh-cn&device_type=pc&ext=%7B%22loc%22%3A%7B%22x%22%3A-476.3782958984375%2C%22y%22%3A210.6436767578125%2C%22z%22%3A330.84381103515627%7D%2C%22platform%22%3A%22WinST%22%7D&game_version=CNRELWin2.4.0_R5578371_S5635986_D5598173&plat_type=pc&region=cn_gf01&authkey=O7lOOv8JUMCje8be10Qtfk%2FbOcb7KvKVzi2dFMgkYjkq4tuk%2BMneQjRIgzvWQm6bHQzbhIWuoLgNamvYhD2FvKhP9bDLyTf9RwkeFDxjT2oJpOsEEEMpeH5baIA4HfuJy%2FGMnPk0b7gXdFV9WRar2XxI%2BPal9tn%2BiZ%2B9vhDGHF%2FE4%2F4CRzgBSxBUnYpRRZYHSKst3CWOSGKiJuGJDWBsbE3wCap%2FS6TeH%2F%2Fj7sgc3Vib1dZW0qUcgA7wPrpRIAyQuj4oqgHGxqJ7XdTxOs%2BwLXuQKBVSqEUGShTvVliWh5etRk6NGDa8Kwo1tJGFbCOdWK%2FvVkadSXN%2FWjosVpq1xaVDQbrTBEeuYSFZJsXEeuXPK%2BChoFI3OZoY9ZHa7cyrpMI3MVyDEUw5mUqtXl%2Fu30depAsglOSs0aLYlqD7EV4mPn4u11mIhi260VA38mSi8YCauCYYASqWL3wF2CdguClzRsfqVOu0CPEJEfgyYXWqe0680f7J7Bpxa1A0kIbQrQVwctIq%2FHrf3JpjNkZz3WxgTxhczGWue4DA2VASFLXmWamnzHNxhTxkGagVHGBTuSxeod7eNrS%2FLR%2F8iOWWhjX3Kl1fofLR5rQU9n4K%2F3g3mEGNMutp1U5jwnTP%2B3%2B19uyyho7fwVsNghP%2Fd%2BK5eAJwtcjb9VijjGdhfUBsVrvPJVukPxfCg7KRpDcxndFS0iFHQ%2FnahAS9V%2F8ztYAFRB4u68c%2F8JaX12H8VtW5LQgnskRNOOMiKwETNGacRY58c1QQG8D56YIgTMWDZF%2FmuWQstrpPxec%2Fj%2BM71sDOmfyGeTGfRMYp7R3MXcSr%2BRu7rA0mj1Z28V%2B1AXjPfMSzoijy6ePkApzdVFDbQ29i%2B3tEPAmz76cOZqUkk%2Fg0TBDKGqEBPPrq7aTDqqu0T3ff0AGQTQExPVvh7k3mD%2FMxTxP5dy9zimM6EJbJ%2BMPreQ0MOBSNwJWjCtHPC7tCCE6sgU0Mg7VgB0qlYMMgvDnQW2%2FayIpAAg%2F5bVbykis74RKyeBpLL85UN3TRbLnOsj%2FqmEakKNwjTwkr1OTkGzchfuVqEvL5UCcAJSdqUja6pQC6KakfvR%2BHMlVZIJPLPHHpewwHz9K%2BMbhbKtMtGmQ6KVY8JpqhZRbZKIIVHLeUWEWjhYxbVjeM6qtX3hGk6FHJmZIxu5HxUFOLYC4Sy45F%2BPMrGjKLpueecdk%2FcRO3v6HQ61q7AOoWiPofHbaJcFn2L563hnKBlY5si5DNzTnGeiOjkxzQWd1l%2FJdjuJx8VjyIjxDQYVEjNfo8ZNmuglq3VDVEA%2FzNw3oFr23ZqvZVxje7lwzE0VQrw5JT2WaB66wT8EgccQrO0%2F3h0Gy%2FWT1MXA%3D%3D&game_biz=hk4e_cn&size=20&gacha_type=301&page=14&end_id=1642741560000896558) 我这边是能打开的，有可能是你网络问题或者代理导致的

**Reply by guwukawa** [reply]
我这边也是能打开的，而且这个错误不是马上出现，是获取到一定进度之后获取进度会卡着，然后才弹出以上错误

**Reply by guwukawa** [reply]
大概的流程如下


检查更新中...
检查配置文件中的链接...合法
获取抽卡记录
正在获取 新手祈愿 第 1 页
正在获取 常驻祈愿 第 1 页
正在获取 常驻祈愿 第 2 页
日志读取模块出错:
 Traceback (most recent call last):
  File "urllib3\connection.py", line 169, in _new_conn
  File "urllib3\util\connection.py", line 96, in create_connection
  File "urllib3\util\connection.py", line 86, in create_connection
TimeoutError: [WinError 10060] 由于连接方在一段时间后没有正确答复或连接的主机没有反应，连接尝试失败。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "urllib3\connectionpool.py", line 699, in urlopen
  File "urllib3\connectionpool.py", line 382, in _make_request
  File "urllib3\connectionpool.py", line 1010, in _validate_conn
  File "urllib3\connection.py", line 353, in connect
  File "urllib3\connection.py", line 181, in _new_conn
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x000001D78F89CA60>: Failed to establish a new connection: [WinError 10060] 由于连接方在一段时间后没有正确答复或连接的主机没有反应，连接尝试失败。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "requests\adapters.py", line 439, in send
  File "urllib3\connectionpool.py", line 755, in urlopen
  File "urllib3\util\retry.py", line 573, in increment
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='hk4e-api.mihoyo.com', port=443): Max retries exceeded with url: /event/gacha_info/api/getGachaLog?authkey_ver=1&sign_type=2&auth_appid=webview_gacha&init_type=301&gacha_id=c62e2e1cce2be432c1e306881b7b9a1ff3bd9c&timestamp=1641338923&lang=zh-cn&device_type=pc&ext=%7B%22loc%22%3A%7B%22x%22%3A-3013.2412109375%2C%22y%22%3A221.2991485595703%2C%22z%22%3A-4348.47119140625%7D%2C%22platform%22%3A%22WinST%22%7D&game_version=CNRELWin2.4.0_R5578371_S5635986_D5598173&plat_type=pc&region=cn_gf01&authkey=O7lOOv8JUMCje8be10Qtfk%2FbOcb7KvKVzi2dFMgkYjkq4tuk%2BMneQjRIgzvWQm6bHQzbhIWuoLgNamvYhD2FvKhP9bDLyTf9RwkeFDxjT2oJpOsEEEMpeH5baIA4HfuJy%2FGMnPk0b7gXdFV9WRar2XxI%2BPal9tn%2BiZ%2B9vhDGHF%2FE4%2F4CRzgBSxBUnYpRRZYHSKst3CWOSGKiJuGJDWBsbE3wCap%2FS6TeH%2F%2Fj7sgc3Vib1dZW0qUcgA7wPrpRIAyQuj4oqgHGxqJ7XdTxOs%2BwLXuQKBVSqEUGShTvVliWh5etRk6NGDa8Kwo1tJGFbCOdWK%2FvVkadSXN%2FWjosVpq1xaVDQbrTBEeuYSFZJsXEeuXPK%2BChoFI3OZoY9ZHa7cyrpMI3MVyDEUw5mUqtXl%2Fu30depAsglOSs0aLYlqD7EV4mPn4u11mIhi260VA38mSi8YCauCYYASqWL3wF2CdguClzRsfqVOu0CPEJEfgyYXWqe0680f7J7Bpxa1A0kIbQrQVwctIq%2FHrf3JpjNkZz3WxgTxhczGWue4DA2VASFLXmWamnzHNxhTxkGagVHGBTuSxeod7eNrS%2FLR%2F8iOWWhjX3Kl1fofLR5rQU9n4K%2F3g3mEGNMutp1U5jwnTP%2B3%2B19uyyho7fwVsNghP%2Fd%2BK5eAJwtcjb9VijjGdhfUBsVrvOFadVOt0fkwWGvsHaH2orZxUk2qp8ln0qZwDBk3EsteI6rQOxjbH%2F2a3EJ%2F0HqNL1mBayVvXnssqWGsDUdNBkqIsWDgyyoYJGET4rYuGSVGjnncRh0XN8Oj7oMF088ESQqxs8FEb483Gbqwj%2BpPjGeQT8YqQxSINDm2I6bHnqe70JwGUfeUvPPP%2Fob8M7d35fmtRizOotBPyHMKrowyCVjdkism%2BX%2BJW0%2FlLv9Jga8JvifcMlwxglTEjK19yJoWaP9mn5DJLBws25JXj1WCa9XYOwm%2F1dyKy8JYBaLjeRNd5tzdM6T9YcgfZI7PqKoKbCvNTpnL3JzLhCBWZL%2BQrEeBpLL85UN3TRbLnOsj%2FqmEakKNwjTwkr1OTkGzchfuVqEvL5UCcAJSdqUja6pQC6KakfvR%2BHMlVZIJPLPHHpewwHz9K%2BMbhbKtMtGmQ6KVY8JpqhZRbZKIIVHLeUWEWjhYxbVjeM6qtX3hGk6FHJmZIxu5HxUFOLYC4Sy45F%2BPMrGjKLpueecdk%2FcRO3v6HQ61q7AOoWiPofHbaJcFn2L563hnKBlY5si5DNzTnGeiOjkxzQWd1l%2FJdjuJx8VjyIjxDQYVEjNfo8ZNmuglq3VDVEA%2FzNw3oFr23ZqvZVxje7lwzE0VQrw5JT2WaB66wT8EgccQrO0%2F3h0Gy%2FWT1MXA%3D%3D&game_biz=hk4e_cn&size=20&gacha_type=200&page=2&end_id=1638468360000192858 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x000001D78F89CA60>: Failed to establish a new connection: [WinError 10060] 由于连接方在一段时间后没有正确答复或连接的主机没有反应，连接尝试失败。'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "main.py", line 414, in <module>
    print("合法")
  File "main.py", line 30, in main
    gachaLog = getGachaLogs(gachaTypeId, gachaQueryTypeDict)
  File "main.py", line 160, in getGachaLogs
    r = requests.get(api)
  File "requests\api.py", line 76, in get
  File "requests\api.py", line 61, in request
  File "requests\sessions.py", line 542, in request
  File "requests\sessions.py", line 655, in send
  File "requests\adapters.py", line 516, in send
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='hk4e-api.mihoyo.com', port=443): Max retries exceeded with url: /event/gacha_info/api/getGachaLog?authkey_ver=1&sign_type=2&auth_appid=webview_gacha&init_type=301&gacha_id=c62e2e1cce2be432c1e306881b7b9a1ff3bd9c&timestamp=1641338923&lang=zh-cn&device_type=pc&ext=%7B%22loc%22%3A%7B%22x%22%3A-3013.2412109375%2C%22y%22%3A221.2991485595703%2C%22z%22%3A-4348.47119140625%7D%2C%22platform%22%3A%22WinST%22%7D&game_version=CNRELWin2.4.0_R5578371_S5635986_D5598173&plat_type=pc&region=cn_gf01&authkey=O7lOOv8JUMCje8be10Qtfk%2FbOcb7KvKVzi2dFMgkYjkq4tuk%2BMneQjRIgzvWQm6bHQzbhIWuoLgNamvYhD2FvKhP9bDLyTf9RwkeFDxjT2oJpOsEEEMpeH5baIA4HfuJy%2FGMnPk0b7gXdFV9WRar2XxI%2BPal9tn%2BiZ%2B9vhDGHF%2FE4%2F4CRzgBSxBUnYpRRZYHSKst3CWOSGKiJuGJDWBsbE3wCap%2FS6TeH%2F%2Fj7sgc3Vib1dZW0qUcgA7wPrpRIAyQuj4oqgHGxqJ7XdTxOs%2BwLXuQKBVSqEUGShTvVliWh5etRk6NGDa8Kwo1tJGFbCOdWK%2FvVkadSXN%2FWjosVpq1xaVDQbrTBEeuYSFZJsXEeuXPK%2BChoFI3OZoY9ZHa7cyrpMI3MVyDEUw5mUqtXl%2Fu30depAsglOSs0aLYlqD7EV4mPn4u11mIhi260VA38mSi8YCauCYYASqWL3wF2CdguClzRsfqVOu0CPEJEfgyYXWqe0680f7J7Bpxa1A0kIbQrQVwctIq%2FHrf3JpjNkZz3WxgTxhczGWue4DA2VASFLXmWamnzHNxhTxkGagVHGBTuSxeod7eNrS%2FLR%2F8iOWWhjX3Kl1fofLR5rQU9n4K%2F3g3mEGNMutp1U5jwnTP%2B3%2B19uyyho7fwVsNghP%2Fd%2BK5eAJwtcjb9VijjGdhfUBsVrvOFadVOt0fkwWGvsHaH2orZxUk2qp8ln0qZwDBk3EsteI6rQOxjbH%2F2a3EJ%2F0HqNL1mBayVvXnssqWGsDUdNBkqIsWDgyyoYJGET4rYuGSVGjnncRh0XN8Oj7oMF088ESQqxs8FEb483Gbqwj%2BpPjGeQT8YqQxSINDm2I6bHnqe70JwGUfeUvPPP%2Fob8M7d35fmtRizOotBPyHMKrowyCVjdkism%2BX%2BJW0%2FlLv9Jga8JvifcMlwxglTEjK19yJoWaP9mn5DJLBws25JXj1WCa9XYOwm%2F1dyKy8JYBaLjeRNd5tzdM6T9YcgfZI7PqKoKbCvNTpnL3JzLhCBWZL%2BQrEeBpLL85UN3TRbLnOsj%2FqmEakKNwjTwkr1OTkGzchfuVqEvL5UCcAJSdqUja6pQC6KakfvR%2BHMlVZIJPLPHHpewwHz9K%2BMbhbKtMtGmQ6KVY8JpqhZRbZKIIVHLeUWEWjhYxbVjeM6qtX3hGk6FHJmZIxu5HxUFOLYC4Sy45F%2BPMrGjKLpueecdk%2FcRO3v6HQ61q7AOoWiPofHbaJcFn2L563hnKBlY5si5DNzTnGeiOjkxzQWd1l%2FJdjuJx8VjyIjxDQYVEjNfo8ZNmuglq3VDVEA%2FzNw3oFr23ZqvZVxje7lwzE0VQrw5JT2WaB66wT8EgccQrO0%2F3h0Gy%2FWT1MXA%3D%3D&game_biz=hk4e_cn&size=20&gacha_type=200&page=2&end_id=1638468360000192858 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x000001D78F89CA60>: Failed to establish a new connection: [WinError 10060] 由于连接方在一段时间后没有正确答复或连 接的主机没有反应，连接尝试失败。'))


执行结束，按任意键退出...

**Reply by guwukawa** [reply]
出现错误的位置不确定且无规律可言，或长或短

**Comment by sunfkny** [comment, marked as answer]
大概率和你网络有关，换个网络试试，报错的部分其实都是标准库urllib3的东西，和我关系不大
你也可以看看 [这段demo](http://www.dooccn.com/python3/#id/51a201628d50692b3519f1e21fb2859c) ，和我代码核心逻辑一模一样，status_code 都是200，最好是自己本地用 python 跑一跑试试，我本地循环跑没有任何问题
